src/common-utils/main.sh

```diff
@@ -427,7 +427,27 @@ if [ "${RC_SNIPPET_ALREADY_ADDED}" != "true" ]; then
 fi
 
 # Optionally configure zsh and Oh My Zsh!
+omz_rc_filename=".zshrc"
+omz_source_dirname=".oh-my-zsh"
+user_rc_file="${user_home}/${omz_rc_filename}"
+user_omz_install_dir="${user_home}/${omz_source_dirname}"
+template_path="${user_omz_install_dir}/templates/zshrc.zsh-template"
+
+# Allow upstream steps to use installOhMyZshConfig false
+# Given previous step configured ~/.zshrc,
+# When installOhMyZshConfig is false, done before INSTALL_ZSH since
+#  Or where installOhMyZshconfig false, and installZsh false
+# Then remove the file
```

Comment:
With the PR changes `installOhMyZshConfig` would behave completely different as compared to other Feature options.

For eg. Let's say an image (say `X`) installs common-utils as follows -

```jsonc
"features": {
            "common-utils": {
                "installZsh": true,
            }
        }
```

Then a user creates their `devcontainer.json` as 👇 

```
{
        "image": "X",
        "features": {
            "common-utils": {
                "installZsh": false
            }
        }
    }
```

> "installZsh": false

Does this mean the Feature should remove the `zsh` installation? I don't think that the Features should remove an existing functionality, however, should only update or add new tools/configs/themes.

When a user add a Feature to their `devcontainer.json`, it is assumed that the `"installZsh": false` option would NOT install `zsh` but it doesn't mean that it should remove the already existing `zsh`. This behavior could create confusions and removing something is always a bit risky. I think we should avoid this approach.

@naturedamends Let me know what you think!

Reply:
Maybe, remove the file, if zshrc exists and installZshConfig false and installZsh true?

Or do you not like the file being removed at all?

Should I remove the source of zshrc added in tests? Such that it's not executed within within build runtime. Then again.. it's a container

Reply:
> Or do you not like the file being removed at all?

This ☝️ 
You never know if the zsh config has been modified by the dockerfiles (or something else), hence, it's best to avoid removing anything. Also, removing config makes sense only if the Feature had an option like  `removeZshConfig `, but doesn't seem like that's needed.

Reply:
Hose about linking ~/.zshrc to .zshrc.devcontainer.

Them downstream can handle, or ln -sf?